src/App.vue

```diff
@@ -78,12 +80,56 @@ export default {
 		}
 	},
 
+	computed: {
+		conversations() {
+			return this.$store.getters.conversations
+		},
+		/**
+		 * The current conversation token
+		 * @returns {string} The token.
+		 */
+		token() {
+			return this.$route.params.token
+		}
+	},
+
 	beforeMount() {
 		window.addEventListener('resize', this.onResize)
 		this.onResize()
+		/**
+		 * Listens to the conversationsReceived globalevent, emitted by the conversationsList
+		 * component each time a new batch of conversations is received and processed in
+		 * the store.
+		 */
+		EventBus.$once('conversationsReceived', () => {
+			if (this.$route.name === 'conversation') {
+				const CURRENT_CONVERSATION_NAME = this.getConversationName(this.token)
+				this.setPageTitle(CURRENT_CONVERSATION_NAME)
+			}
+		})
+		/**
+		 * Global before guard, this is called whenever a navigation is triggered.
+		*/
+		Router.beforeEach((to, from, next) => {
+			/**
+			 * This runs whenever the new route is a conversation.
+			 */
+			if (to.name === 'conversation') {
+				const NEXT_CONVERSATION_NAME = this.getConversationName(to.params.token)
+				this.setPageTitle(NEXT_CONVERSATION_NAME)
+			}
+		})
 	},
 
 	methods: {
+		/**
+		 * Set the page title to the conversation name
+		 * @param {string} title Prefix for the page title e.g. conversation name
+		 */
+		setPageTitle(title) {
+			window.document.title = `${title} - ${t('spreed', 'Talk')}`
```

Comment:
```js
			if (this.defaultPageTitle === false) {
				// On the first load we store the current page title "Talk - Nextcloud",
				// so we can append it every time again
				this.defaultPageTitle = window.document.title
				// When a conversation is opened directly, the "Talk - " part is
				// missing from the title
				if (this.defaultPageTitle.indexOf(t('spreed', 'Talk') + ' - ') !== 0) {
					this.defaultPageTitle = t('spreed', 'Talk') + ' - ' + this.defaultPageTitle
				}
			}
```
Needs to be brought back, otherwise Nextcloud is removed from the title when you navigate the first time